src/autopas/molecularDynamics/ParticlePropertiesLibrary.h

```diff
@@ -0,0 +1,152 @@
+  using ParticlePropertiesLibraryFloatType = floatType;
+
+  /**
+   * Type for integer numbers.
+   */
+  using ParticlePropertiesLibraryIntType = intType;
+
+  /**
+   * Default constructor.
+   */
+  ParticlePropertiesLibrary() = default;
+
+  /**
+   * Copy Constructor.
+   * @param particlePropertiesLibrary
+   */
+  ParticlePropertiesLibrary(const ParticlePropertiesLibrary &particlePropertiesLibrary) = default;
+
+  /**
+   * Copy assignment operator.
+   * @param particlePropertiesLibrary
+   * @return
+   */
+  ParticlePropertiesLibrary &operator=(const ParticlePropertiesLibrary &particlePropertiesLibrary) = default;
+
+  /**
+   * Adds the properties of a particle type to the library.
+   *
+   * This function also precomputes all possible mixed values with already known particle types.
+   * If the type id already exists the values will be overwritten.
+   * @param typeID
+   * @param epsilon
+   * @param sigma
+   * @param mass
+   */
+  void addType(intType typeID, floatType epsilon, floatType sigma, floatType mass);
+
+  ~ParticlePropertiesLibrary() = default;
+
+  /**
+   * Getter for the particle's epsilon*24.
+   * @param i typeId of the particle.
+   * @return 24*epsilon_i
+   */
+  floatType get24Epsilon(intType i);
+
+  /**
+   * Getter for the particle's squared sigma.
+   * @param i typeId of the particle.
+   * @return sigma_i²
+   */
+  floatType getSigmaSquare(intType i);
+
+  /**
+   * Getter for the particle's mass.
+   * @param i typeId of the particle.
+   * @return mass_i
+   */
+  floatType getMass(intType i);
+
+  /**
+   * Returns the precomputed mixed epsilon24.
+   * @param  i typeId index of particle one.
+   * @param  j typeId index of particle two.
+   * @return 24*epsilon_ij
+   * */
+  inline floatType mixing24Epsilon(intType i, intType j) const {
+    auto key = std::make_pair((i < j) ? i : j, (j > i) ? j : i);  // key in preprocessed maps: (i,j) with i<j
+    return _computedMixing24Epsilon.at(key);
+  }
+  /**
+   * Returns precomputed mixed squared sigma.
+   * @param i typeId index of particle one.
+   * @param j typeId index of particle two.
+   * @return sigma_ij²
+   */
+  inline floatType mixingSigmaSquare(intType i, intType j) const {
+    auto key = std::make_pair((i < j) ? i : j, (j > i) ? j : i);  // key in preprocessed maps: (i,j) with i<j
+    return _computedMixingSigmaSquare.at(key);
+  }
+
+ private:
+  std::map<intType, floatType> _epsilons;
+  std::map<intType, floatType> _sigmas;
+  std::map<intType, floatType> _masses;
+  std::map<std::pair<intType, intType>, floatType> _computedMixing24Epsilon;
+  std::map<std::pair<intType, intType>, floatType> _computedMixingSigmaSquare;
+};
+
+template <typename floatType, typename intType>
+void ParticlePropertiesLibrary<floatType, intType>::addType(intType typeID, floatType epsilon, floatType sigma,
+                                                            floatType mass) {
+  _epsilons.emplace(typeID, epsilon);
+  for (auto &e : _epsilons) {
```

Comment:
C++17 structured binding is great for map iterations and makes the code clearer. Here, it would be something like
```
for(auto const & [indexOfExistingEpsilon, secondEpsilon] : _epsilons)
```